h2o-algos/src/main/java/hex/glrm/GLRM.java

```diff
@@ -241,16 +245,26 @@ private void initLoss() {
     for (int i = 0; i < _ncolA; i++) {
       Vec vi = _train.vec(i);
       GlrmLoss lossi = _lossFunc[i];
-      if (vi.isNumeric()) {
-        if (!lossi.isForNumeric())
-          error("_loss_by_col", "Loss function " + lossi + " cannot be applied to numeric column " + i);
-      } else if (vi.isCategorical()) {
-        if (!lossi.isForCategorical())
-          error("_loss_by_col", "Loss function " + lossi + " cannot be applied to categorical column " + i);
-      } else if (vi.isBinary()) {
-        // Allow numeric loss functions on binary columns too; but not the other way round!
-        if (!lossi.isForBinary() && !lossi.isForNumeric())
-          error("_loss_by_col", "Loss function " + lossi + " cannot be applied to binary column " + i);
+
+      if (vi.isNumeric())   { // numeric columns
+        if (!vi.isBinary()) { // non-binary numeric columns
+          if (!lossi.isForNumeric())
+            error("_loss_by_col", "Loss function "+lossi+" cannot be applied to numeric column "+i);
+        } else {  // binary numeric columns
```

Comment:
In this case the column vi is already numerical.  There is no need to change to numerical later.